docs/user-guide/sns-instruments/vanadium_processing.ipynb

```diff
@@ -0,0 +1,381 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "fbcde35f-8608-4ecb-9b6d-67f686d7839a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "incoherent = scn.peaks.remove_peaks(sc.values(to_fit), fit_results)\n",
+    "incoherent.coords['dspacing'] = peak_histogram.coords['dspacing']\n",
+    "incoherent.plot()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6f19a754-c41a-4507-9ac5-a9f68dbab60f",
+   "metadata": {},
+   "source": [
+    "We can further inspect the results.\n",
+    "Below, there is a function that plots\n",
+    "\n",
+    "- the data with coherent and incoherent scattering (blue),\n",
+    "- the resulting incoherent curve (green),\n",
+    "- the fitted models (orange),\n",
+    "- the fit windows (gray and red boxes),\n",
+    "- and the initial estimates (dashed vertical lines).\n",
+    "\n",
+    "Some fits failed as indicated by red boxes and short descriptions of why the fits failed.\n",
+    "Some peaks are absent from the data used here, even though they are expected based on the crystal structure.\n",
+    "So those fits are expected to fail.\n",
+    "All other fits appear to have succeeded.\n",
+    "\n",
+    "See [scippneutron.peaks.fit_peaks](https://scipp.github.io/scippneutron/generated/modules/scippneutron.peaks.fit_peaks.html) for options to customize the fit procedure if it does not work as desired.\n",
+    "\n",
+    "<div class=\"alert alert-info\">\n",
+    "\n",
+    "**Note**\n",
+    "\n",
+    "It is highly recommended to inspect the plot in detail to check whether all fits have succeeded or failed as expected!\n",
+    "Fitting is not always reliable and may fail for many reasons.\n",
+    "You can make plots interactive by using `%matplotlib widget`.\n",
+    "\n",
+    "</div>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "0fd4e65a-35b3-4c12-a4ee-24649bda2551",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def peak_removal_diagnostic(\n",
+    "    data: sc.DataArray,\n",
+    "    removed: sc.DataArray,\n",
+    "    fit_results: list[scn.peaks.FitResult],\n",
+    "    peak_estimates: sc.Variable,\n",
+    "    *,\n",
+    "    xlim: tuple[sc.Variable, sc.Variable] | None=None,\n",
+    "):\n",
+    "    if xlim is not None:\n",
+    "        def in_range(x: sc.Variable) -> bool:\n",
+    "            return sc.isfinite(x) and (xlim[0] <= x) and (x < xlim[1])\n",
+    "        data = data[data.dim, xlim[0]:xlim[1]]\n",
+    "        removed = removed[removed.dim, xlim[0]:xlim[1]]\n",
+    "        fit_results, peak_estimates = zip(*(\n",
+    "            (r, e)\n",
+    "            for r, e in zip(fit_results, peak_estimates, strict=True)\n",
+    "            if in_range(r.window[0]) and in_range(r.window[1])\n",
+    "        ), strict=True)\n",
```

Comment:
I changed this because the behaviour suddenly changed on my machine. I don't know what caused it.
Now, when using `ax.set_xlim` the text shown with `ax.text` is still shown. This messes up how the axes are shown in a figure. The new solution removes everything that is not within the shown range. It may remove a fit that lies at the edge of the range, though.